k5test/unit.py

```diff
@@ -35,15 +35,20 @@ def ext_test(self, *args, **kwargs):
 _KRB_VERSION = None
 
 
-def krb_minversion_test(target_version, problem):
+def krb_minversion_test(target_version, problem, provider=None):
     global _KRB_VERSION
     if _KRB_VERSION is None:
         _KRB_VERSION = _utils.get_output("krb5-config --version")
         _KRB_VERSION = _KRB_VERSION.split(' ')[-1].split('.')
 
     def make_ext_test(func):
         def ext_test(self, *args, **kwargs):
-            if _KRB_VERSION < target_version.split('.'):
+            if (
+                not provider or (
+                    self.realm.provider.lower() == provider.lower()
+                ) and
+                _KRB_VERSION < target_version.split('.')
+            ):
```

Comment:
Same deal as above, but this one's even more confusing because as written it requires knowing the relative precedence of `or` and `and`.  If memory serves, this would be equivalent to:

```python
            if not provider or (
                    self.realm.provider.lower() == provider.lower() and
                    _KRB_VERSION < target_version.split(".")):
```

Reply:
Hmm even then I think the logic was incorrect here. It should be that if the provider was specified then the version check only applies to that version. Would allow you to do something like the below where it checks the version for the specified provider or all if not specified.

```python
@krb5_min_version_test('1.1', 'some problem with MIT', provider='mit')
@krb5_min_version_test('7.7', 'some problem with Heimdal', provider='heimdal')
```

I've changed it to

```python
if _KRB_VERSION < target_version.split('.') and (
    not provider or self.realm.lower() == provider.lower()
):
```